src/CommunityToolkit.Maui.Markup/TypedBindingExtensions.cs

```diff
@@ -0,0 +1,200 @@
+﻿using System.Linq.Expressions;
+using Microsoft.Maui.Controls.Internals;
+
+namespace CommunityToolkit.Maui.Markup;
+
+/// <summary>
+/// TypedBinding Extension Methods for Bindable Objects
+/// </summary>
+public static class TypedBindingExtensions
+{
+	/// <summary>Bind to a specified property</summary>
+	public static TBindable Bind<TBindable, TBindingContext, TSource>(
+		this TBindable bindable,
+		BindableProperty targetProperty,
+		Expression<Func<TBindingContext, TSource>> getter,
+		Action<TBindingContext, TSource>? setter = null,
+		BindingMode mode = BindingMode.Default,
```

Comment:
Hmmm...yea, I guess that would be a valid input.

I'll add a unit test today to test it out. Depending on how it does/doesn't work, I may add an `ArgumentException`.

```cs
if(setter is not null && mode is not BindingMode.TwoWay)
{
  throw new ArgumentException("...");
}
```

`BindingMode.Default` is `BindingMode.TwoWay` for the controls that support it, so I should also check the control's default binding too

Reply:
Or maybe we just accept that the consumer knows what they are doing around this area? Bindings let you set them up incorrectly as they currently stand.

Reply:
Hey Shaun! I was just playing around with this argument validation idea, and after giving it some thought, I think it's best for us to pass through the user's inputs to .NET MAUI.

If the .NET MAUI team was to do argument validation, they're welcome to, but in this case, since `BindableObject.SetBinding()` allows these combination of values to be set, we should honor that.

Plus, if we do argument validation, then we risk a breaking change to `CommunityToolkit.Maui.Markup` whenever the .NET MAUI team updates the binding engine.

For posterity, here's the code I wrote to validate the inputs if we ever change our mind and decide to add it to our TypedBindings:

```cs
if (setter is not null)
{
	if (mode is not BindingMode.TwoWay or BindingMode.OneWayToSource)
	{
		throw new ArgumentException($"""
Invalid {nameof(BindingMode)}.
A {nameof(BindingMode)} of {mode} cannot be used to set the value of a property on the binding source (typically a View Model).
Either remove the {nameof(setter)} (changing it to null), or set {nameof(BindingMode)} {nameof(mode)} to either {nameof(BindingMode.TwoWay)} or {nameof(BindingMode.OneWayToSource)}
""", nameof(mode));
	}

	if (mode is BindingMode.Default && targetProperty.DefaultBindingMode is not BindingMode.TwoWay or BindingMode.OneWayToSource)
	{
		throw new ArgumentException($"""
A {nameof(setter)} cannot be applied because the {nameof(BindingMode.Default)} is {targetProperty.DefaultBindingMode}.
Either remove the {nameof(setter)} (changing it to null), or change {nameof(BindingMode)} {nameof(mode)} to either {nameof(BindingMode.TwoWay)} or {nameof(BindingMode.OneWayToSource)}
""", nameof(mode));
	}
}
```